core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -78,7 +78,20 @@ class GroupMetadataManager(brokerId: Int,
   private val shuttingDown = new AtomicBoolean(false)
 
   /* number of partitions for the consumer metadata topic */
-  private val groupMetadataTopicPartitionCount = getGroupMetadataTopicPartitionCount
+  private var _groupMetadataTopicPartitionCount: Option[Int] = Option.empty // lazy, once-only evaluation
+  private def groupMetadataTopicPartitionCount: Int = {
+    _groupMetadataTopicPartitionCount match {
+      case Some(partitionCount) => partitionCount
+      case None => synchronized { // make sure we only invoke the function once
+        _groupMetadataTopicPartitionCount match {
+          case Some(partitionCount) => partitionCount // another thread beat us to it
+          case None =>
+            _groupMetadataTopicPartitionCount = Some(offsetsTopicPartitionCountFunc())
```

Comment:
It's worth noting that if our underlying function throws an error, this will throw as well for any usages of `groupMetadataTopicPartitionCount` (e.g., `GroupMetadataManager#partitionFor`). That wasn't possible in the ZK world (since this value was cached on startup), so we should probably make sure the callers can all tolerate an exception.

Reply:
This is no longer na issue since we invoke the function once in `startup()`.  We do it again in `TransactionStateManager` whenever we become leader for a partition, but that happens in the scheduler thread where any exception will be caught/logged.